src/components/spreadsheet/custom-columns/custom-columns-nodes-config.tsx

```diff
@@ -20,31 +27,125 @@ const styles = {
+
+    const handleRefresh = useCallback(() => {
+        if (nodesToReload?.length) {
+            const nodesIdsToReload = new Set<string>(nodesToReload.map((n) => n.id as string));
+            fetchNodesEquipementData(nodesIdsToReload, undefined);
+        }
+    }, [fetchNodesEquipementData, nodesToReload]);
+
+    const nodesOptions = useMemo(
+        () => ({
+            [NodesOptionId.CONFIG]: {
+                id: NodesOptionId.CONFIG,
+                label: 'spreadsheet/custom_column/option/parameter',
+                action: dialogOpen.setTrue,
+            },
+            [NodesOptionId.REFRESH]: {
+                id: NodesOptionId.REFRESH,
+                label: 'spreadsheet/custom_column/option/refresh',
+                action: () => handleRefresh(),
+                disabled: nodesToReload ? nodesToReload.length === 0 : true,
+                tooltipMsgId: 'spreadsheet/custom_column/option/refresh/tooltip',
+                tooltipMsgValues: {
+                    aliases: nodesToReload?.map((node) => node.alias).join(', '),
+                },
+            },
+        }),
+        [dialogOpen.setTrue, nodesToReload, handleRefresh]
+    );
+
+    const handleMenuItemClick = useCallback(
+        (optionId: NodesOptionId) => {
+            nodesOptions[optionId].action();
+            handleClose();
+        },
+        [nodesOptions, handleClose]
+    );
+
+    const renderMenuItem = useCallback(
+        (option: NodesOption) => {
+            return (
+                <MenuItem key={option.id} onClick={() => handleMenuItemClick(option.id)} disabled={option?.disabled}>
+                    <FormattedMessage id={option.label} />
+                </MenuItem>
+            );
+        },
+        [handleMenuItemClick]
+    );
+
+    const renderMenuEntry = useCallback(
+        (option: NodesOption) => {
+            return (
+                <>
+                    {option.tooltipMsgId && (
+                        <Tooltip title={<FormattedMessage id={option.tooltipMsgId} values={option.tooltipMsgValues} />}>
+                            {renderMenuItem(option)}
+                        </Tooltip>
+                    )}
+                    {!option.tooltipMsgId && renderMenuItem(option)}
+                </>
+            );
+        },
+        [renderMenuItem]
+    );
 
     return (
         <>
-            <Button
-                sx={spreadsheetStyles.spreadsheetButton}
-                size={'small'}
-                onClick={dialogOpen.setTrue}
-                disabled={disabled}
-            >
+            <Button sx={spreadsheetStyles.spreadsheetButton} size={'small'} onClick={handleClick} disabled={disabled}>
                 <BuildIcon sx={styles.icon} />
                 <FormattedMessage id="spreadsheet/custom_column/nodes" />
             </Button>
-
+            <Menu anchorEl={anchorEl} open={Boolean(anchorEl)} onClose={handleClose}>
+                {Object.values(nodesOptions).map(renderMenuEntry)}
```

Comment:
```suggestion
               <MenuItem key={NodesOptionId.CONFIG} onClick={dialogOpen.setTrue}>
                    <FormattedMessage id={'spreadsheet/custom_column/option/parameter'} />
                </MenuItem>
                <Tooltip
                    title={
                        <FormattedMessage
                            id={'spreadsheet/custom_column/option/refresh/tooltip'}
                            values={{
                                aliases: nodesToReload?.map((node) => node.alias).join(', '),
                            }}
                        />
                    }
                >
                    <MenuItem
                        key={NodesOptionId.REFRESH}
                        onClick={handleRefresh}
                        disabled={nodesToReload ? nodesToReload.length === 0 : true}
                    >
                        <FormattedMessage id={'spreadsheet/custom_column/option/refresh/tooltip'} />
                    </MenuItem>
                </Tooltip>
```
Can we just define the components directly, all the dynamic thing is verbose and maybe not really useful

Reply:
yes for 2 items it does not worth it